src/client/signatures/openssl_verifier.rs

```diff
@@ -0,0 +1,81 @@
+#[cfg(feature = "openssl_verifier")]
+use {
+    super::{Collection, SignatureError, Verification},
+    crate::client::signatures::canonical_json::serialize,
+    async_trait::async_trait,
+    log::debug,
+    openssl::bn::BigNumContext,
+    openssl::ec::{EcGroup, PointConversionForm},
+    openssl::nid::Nid,
+    openssl::x509::X509,
+    serde_json::json,
+    signatory::{
+        ecdsa::{curve::NistP384, FixedSignature},
+        verify_sha384, EcdsaPublicKey, Signature,
+    },
+    signatory_ring::ecdsa::P384Verifier,
+};
+
+pub struct OpenSSLVerifier {}
+
+#[cfg(feature = "openssl_verifier")]
+impl From<signatory::error::Error> for SignatureError {
+    fn from(err: signatory::error::Error) -> Self {
+        SignatureError::VerificationError {
+            name: err.to_string(),
+        }
+    }
+}
+
+#[cfg(feature = "openssl_verifier")]
+impl From<openssl::error::ErrorStack> for SignatureError {
+    fn from(err: openssl::error::ErrorStack) -> Self {
+        err.into()
+    }
+}
+
+#[cfg(feature = "openssl_verifier")]
+#[async_trait]
+impl Verification for OpenSSLVerifier {
+    async fn verify(&self, collection: &Collection) -> Result<(), SignatureError> {
```

Comment:
Factor this method, please: metadata -> String -> PEM -> pubkey, pubkey + String -> signature, signature -> error or success.

Reply:
done